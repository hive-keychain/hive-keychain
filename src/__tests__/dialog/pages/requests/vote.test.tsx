import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render, waitFor } from '@testing-library/react';
import React from 'react';
import Vote from 'src/dialog/pages/requests/vote';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestVote from 'src/__tests__/utils-for-testing/data/props/dialog/request-vote';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestVote } from 'src/__tests__/utils-for-testing/types/props-types';
describe('vote tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestVote;
  methods.config();
  it('Must show Vote dialog', async () => {
    const { asFragment } = render(<Vote {...props} />);
    await waitFor(() => {});
    expect(asFragment()).toMatchSnapshot('Request Vote');
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<Vote {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    const clonedProps = objects.clone(props) as PropsRequestVote;
    render(<Vote {...{ ...clonedProps, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    const spySendMessage = jest.spyOn(chrome.runtime, 'sendMessage');
    render(<Vote {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spySendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
