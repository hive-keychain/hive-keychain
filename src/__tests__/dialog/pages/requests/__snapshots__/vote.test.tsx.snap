// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vote tests:
 Must show Vote dialog: Request Vote 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Vote
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Author
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @theghost1980
      </div>
      <h3>
        Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        https://link.hive
      </div>
      <h3>
        Weight
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        1%
      </div>
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to send vote transactions from the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
